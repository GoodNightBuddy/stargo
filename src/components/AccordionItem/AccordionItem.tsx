import React, { FC } from 'react';
import { Accordion } from 'react-bootstrap';
import { IUser } from '../../store/users/types';

interface IAccordionItemProps {
  user: IUser
}

const AccordionItem: FC<IAccordionItemProps> = (props) => {

  return (
    <Accordion.Item eventKey={props.user.id}>
      <Accordion.Header>{props.user.firstName}</Accordion.Header>
      <Accordion.Body>
        <p>First name: {props.user.firstName}</p>
        <p>Last name: {props.user.lastName}</p>
        <p>Age: {props.user.age}</p>
        <p>Gender: {props.user.gender}</p>
        <p>Country: {props.user.country}</p>
      </Accordion.Body>
    </Accordion.Item>
  );
};

export default AccordionItem;