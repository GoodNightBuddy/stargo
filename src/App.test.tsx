import React from 'react';
import { render, screen } from '@testing-library/react';
import { Accordion } from 'react-bootstrap';
import AccordionItem from './components/AccordionItem/AccordionItem';

const mockUsers = [{
  age: 88,
  country: "United States",
  firstName: "Todd",
  gender: "Male",
  id: "93fd4461-2639-41e0-a281-fdad99936d12",
  lastName: "Smith"
}]

describe('AccordionItem component', () => {
  test('Render all user fields', () => {
    render (
      <Accordion className="mx-auto p-4 container">
        {mockUsers.map(user => <AccordionItem user={user} key={user.id}/>)}
      </Accordion>
    );

    expect(screen.getByText('Age: 88')).toBeInTheDocument();
    expect(screen.getByText('First name: Todd')).toBeInTheDocument();
    expect(screen.getByText('Last name: Smith')).toBeInTheDocument();
    expect(screen.getByText('Country: United States')).toBeInTheDocument();
    expect(screen.getByText('Gender: Male')).toBeInTheDocument();

  });

  test('User info snapshot', () => {
    const view = render (
      <Accordion className="mx-auto p-4 container">
        {mockUsers.map(user => <AccordionItem user={user} key={user.id}/>)}
      </Accordion>
    );

    expect(view).toMatchSnapshot();
  })

});
