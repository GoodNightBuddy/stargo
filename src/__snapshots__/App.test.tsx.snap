// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionItem component User info snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mx-auto p-4 container accordion"
      >
        <div
          class="accordion-item"
        >
          <h2
            class="accordion-header"
          >
            <button
              aria-expanded="false"
              class="accordion-button collapsed"
              type="button"
            >
              Todd
            </button>
          </h2>
          <div
            class="accordion-collapse collapse"
          >
            <div
              class="accordion-body"
            >
              <p>
                First name: 
                Todd
              </p>
              <p>
                Last name: 
                Smith
              </p>
              <p>
                Age: 
                88
              </p>
              <p>
                Gender: 
                Male
              </p>
              <p>
                Country: 
                United States
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mx-auto p-4 container accordion"
    >
      <div
        class="accordion-item"
      >
        <h2
          class="accordion-header"
        >
          <button
            aria-expanded="false"
            class="accordion-button collapsed"
            type="button"
          >
            Todd
          </button>
        </h2>
        <div
          class="accordion-collapse collapse"
        >
          <div
            class="accordion-body"
          >
            <p>
              First name: 
              Todd
            </p>
            <p>
              Last name: 
              Smith
            </p>
            <p>
              Age: 
              88
            </p>
            <p>
              Gender: 
              Male
            </p>
            <p>
              Country: 
              United States
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
